test/arrayops.jl

```diff
@@ -797,6 +797,14 @@ end
     oa = OffsetVector(copy(a), -1)
     @test circshift!(oa, 1) === oa
     @test oa == circshift(OffsetVector(a, -1), 1)
+
+    # 1d circshift! (https://github.com/JuliaLang/julia/pull/53554)
+    a = []
+    @test circshift!(a, 1) === a
+    @test circshift!(a, 1) == []
+    a = [1:5;]
+    @test circshift!(a, 10) === a
+    @test circshift!(a, 10) == [1:5;]
```

Comment:
This test seems unnecessary, as the previous one would ensure that the values match?

Reply:
Without the proposed changes each of the four added tests fails. The previous tests don't cover the case where the length of the vector is zero or where it is divisible by the shift amount.

Reply:
Sorry, I meant that `circshift!(a, 10) === a` should already ensure that the values are equal, so the test `circshift!(a, 10) == [1:5;]` seems redundant? The first test is certainly necessary.

Reply:
Okay, thanks for the clarification. The first test merely checks that the identical vector is returned. But this doesn't guarantee that its content hasn't changed (e.g. it could have been shifted an incorrect number of times). Hence, the second test.

Reply:
Oh, you're right